### PR TITLE
Measure the ACPI table generation commands and the physical memory map (E820 table)

### PR DESCRIPTION
**Pull request by conradgrobler**
These are the final missing measurements we needed for attestation evidence.

This change also includes some refactoring to make some of the other measurement code more consistent.

Ref #4074

### CODE REVIEW

File: stage0/src/lib.rs [hunk 1]
```diff
@@ -252,8 +258,12 @@ pub fn rust64_start(encrypted: u64) -> ! {
         entry = VirtAddr::new(header.e_entry);
     }
 
-    let rsdp = acpi::build_acpi_tables(&mut fwcfg).unwrap();
+    let mut acpi_digest = Sha256::default();
```
Comment (ipetr0v) [inline stage0/src/lib.rs:261]: Why are ACPI tables measured separately?

File: stage0/src/lib.rs [hunk 2]
```diff
@@ -252,8 +258,12 @@ pub fn rust64_start(encrypted: u64) -> ! {
...
+    let rsdp = acpi::build_acpi_tables(&mut fwcfg, &mut acpi_digest).unwrap();
```
Comment (ipetr0v) [inline stage0/src/lib.rs:262]: Is Stage0 creating ACPI tables?

**Review by ipetr0v — APPROVED** [review]

File: stage0/src/lib.rs (hunk 1 above)
Comment (conradgrobler) [reply stage0/src/lib.rs:261]: Stage0 constructs ACPI tables by reading multiple files and command over the fw_cfg device from the VMM. The number of files and commands can change, so this keeps extending the single digest for all the files and commands.

File: stage0/src/lib.rs (hunk 2 above)
Comment (conradgrobler) [reply stage0/src/lib.rs:262]: Sort-of. We get files and command from the VMM via the fw_cfg device to construct them. We don't build them fully.

**Pull request merged**